cou/steps/analyze.py

```diff
@@ -111,4 +143,9 @@ def current_cloud_os_release(self) -> Optional[OpenStackRelease]:
         :return: OpenStack release codename
         :rtype: OpenStackRelease
         """
-        return min((app.current_os_release for app in self.apps), default=None)
+        control_plane = (app for app in self.apps_control_plane)
+        all_apps = chain(control_plane, self.apps_data_plane)
```

Comment:
I don't understand why you need the generator expression and chain here. Can `apps_control_plane` and `apps_data_plane` be anything else than lists?

Reply:
they are `Iterable`
Would you prefer something like this:
```
        return min(
            (app.current_os_release for app in [*self.apps_control_plane, *self.apps_data_plane]),
            default=None,
        )
```